client/blocks/privacy-policy-banner/index.jsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getPreference, isFetchingPreferences } from 'state/preferences/selectors';
+import { savePreference } from 'state/preferences/actions';
+import { identity } from 'lodash';
+import Banner from 'components/banner';
+import config from 'config';
+import PrivacyPolicyDialog from './privacy-policy-dialog';
+import QueryPrivacyPolicy from 'components/data/query-privacy-policy';
+import { getPrivacyPolicyByEntity } from 'state/selectors';
+import { AUTOMATTIC_ENTITY, PRIVACY_POLICY_PREFERENCE } from './constants';
+
+class PrivacyPolicyBanner extends Component {
+	static propTypes = {
+		isPolicyAlreadyAccepted: PropTypes.bool,
+		privacyPolicy: PropTypes.object,
+		privacyPolicyId: PropTypes.string,
+		text: PropTypes.string,
+		translate: PropTypes.func,
+	};
+
+	static defaultProps = {
+		privacyPolicy: {},
+		text: '',
+		translate: identity,
+	};
+
+	state = {
+		showDialog: false,
+	};
+
+	acceptUpdates = () => {
+		if ( ! this.props.privacyPolicyId ) {
+			return;
+		}
+
+		const { privacyPolicyId, privacyPolicyUserStatus } = this.props;
+		this.props.acceptPrivacyPolicy( privacyPolicyId, privacyPolicyUserStatus );
+	};
+
+	openPrivacyPolicyDialog = () => this.setState( { showDialog: true } );
+
+	closePrivacyPolicyDialog = () => {
+		this.setState( { showDialog: false } );
+		this.acceptUpdates();
+	};
+
+	render() {
+		if ( ! config.isEnabled( 'privacy-policy' ) ) {
+			return null;
+		}
+
+		const {
+			fetchingPreferences,
+			isPolicyAlreadyAccepted,
+			moment,
+			privacyPolicy,
+			translate,
+		} = this.props;
+
+		if ( fetchingPreferences ) {
+			return null;
+		}
+
+		// check if the user has already accepted/read the privacy policy.
+		if ( isPolicyAlreadyAccepted === true && ! config.isEnabled( 'privacy-policy/test' ) ) {
```

Comment:
If the feature flag isn't enabled, wouldn't we return null instead?

Reply:
We have two testing flags. The first one is the feature one named `privacy-policy`. It's used above.
But in this line, we use `privacy-policy/test` to force to show the banner beyond if the user has already accepted/seen the dialog?
You could modify the endpoint version but it isn't practical to do.

Reply:
btw `config.isEnabled( 'privacy-policy/test' )` will be removed once it's landed.